lib/karafka/processing/worker.rb

```diff
@@ -52,19 +55,21 @@ def process
           Karafka.monitor.instrument('worker.process', instrument_details)
 
           Karafka.monitor.instrument('worker.processed', instrument_details) do
-            job.before_call
+            external_worker_executor.call do
```

Comment:
I'm not sure if this is the best place to integrate it, but it works! Originally, I had it wrapping this entire method including the `job = @jobs_queue.pop` line, and that caused what I assume are threadsafety errors, something about adding to a hash we are iterating.

Reply:
that may be expected. The queue is multi-threaded and locks from multiple threads so I can expect it may cause some issues to the reloaded